src/urbit/app/chess.hoon

```diff
@@ -706,11 +707,10 @@
       ?+  -.sign  (on-agent:default wire sign)
         %poke-ack
           ?~  p.sign
-            ::  if our poke was successful,
-            ::  do nothing
-            ::[~ this]
+            ::  if opponent has acked our acceptance/rejection,
+            ::  confirm that to our frontend
             :-  :~  :*  %give  %fact   ~[/challenges]
-                        %chess-update  !>([%challenge-sent src.bowl])
+                        %chess-update  !>([%challenge-resolved src.bowl])
                     ==
                 ==
             this
```

Comment:
@ashelkovnykov Should the new `%challenge-sent` card go here, so that it's not being added to `challenges-sent` until the ack comes in from the ship we've challenged? If so, how will I access the `chess-challenge` info that needs to be sent to the frontend alongside the opponent ship's `@p`?

Reply:
That's a good point. What I'm thinking:
- Split the poke acks into two wires: `[%poke %challenge %send ~]` and `[%poke %challenge %resolve ~]`
- When we send a challenge in `on-poke`, add it to the set of outgoing challenges (looks like your current code still does). If the request is nacked, remove it from the set of outgoing challenges. If it the request is acked, lookup the `challenge` data by the `@p` of the opponent in the outgoing challenges set, and send that in the update card.

Reply:
Sounds good, I was wondering if the solution was something like that second point.